### PR TITLE
Add beta probability distribution

### PR DESCRIPTION
**Pull request by jmh530**
I make use of the `std.mathspecial` functions from phobos as I have no special insight on how to implement these any differently (very similar to the approach used in dstats, [here](https://github.com/DlangScience/dstats/blob/master/source/dstats/distrib.d)).

### CODE REVIEW

File: source/mir/math/internal/logBeta.d [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/++
+License: $(HTTP www.apache.org/licenses/LICENSE-2.0, Apache-2.0)
+
+Authors: John Michael Hall
+
+Copyright: 2022 Mir Stat Authors.
++/
+
+module mir.math.internal.logBeta;
```
Comment (9il) [inline source/mir/math/internal/logBeta.d]: `log_beta` please

**Comment by 9il** [comment]
~~Can some internal function be replaced with `stdc.tgmath`? If so, please replace~~

**Comment by jmh530** [comment]
@9il `tgmath` has `log1p` that can be used. I have replaced instances of `log(1 - p)` with `log1p(-x)`. `tgmath` also has `lgamma` instead of the `logGamma` in `std.mathspecial`, but I didn't make that change because I noticed your comment had been crossed out.

**Comment by jmh530** [comment]
Oh, and using the `stdc` version requires dropping `pure`

**Comment by jmh530** [comment]
I could also replace the `core.stdc.tgmath` `log1p` with the one from `std.math.exponential`. This would enable us to bring back `pure`.

File: source/mir/math/internal/logBeta.d (hunk 1 above)
Comment (9il) [reply source/mir/math/internal/logBeta.d]: @jmh530 Sorry, I haven't been clear enough. The `logBeta` name for functions is gooid. `log_beta`  is for module only.

**Comment by 9il** [comment]
> I could also replace the `core.stdc.tgmath` `log1p` with the one from `std.math.exponential`. This would enable us to bring back `pure`.

Yes please.

**Comment by jmh530** [comment]
@9il Thanks. Latest changes fix that

**Pull request merged**